core/src/main/scala/org/apache/spark/executor/ProcessTreeMetrics.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.executor
+
+private[spark] trait ProcessTreeMetrics {
```

Comment:
imo, its not worth creating the trait now when there is only one implementation.  We can add that abstraction later if its useful.  (this isn't exposed to users at all anyway.)

If we are keeping this, some of these should be private.

Reply:
I was thinking about future, when someone want to add new implementation, so that they use the same methods and not having differences across platforms

Reply:
yes I understand that goal, but who knows when / if that will ever happen.  in the meantime, it just makes this a slightly harder to follow.  Whenever someone does want to put in another implementation, its pretty easy for them to add the interface at that time.